### DISCUSSION TITLE
Addition of new plots to the family of `lets-plot` `LayerSpec`s

### DISCUSSION THREAD

**Discussion by OSuwaidi** [JetBrains/lets-plot#952] [Ideas]
Thanks for the wonderful work guys, this visualization library is superb!

Are you guys planning to expand the list of supported plots (`LayerSpec`s)? If so, could we suggest/vote on which additional plots we would like to be introduced to the library?

Charts like waterfall, parallel coordinates, and radar plots are all wonderful ideas!

**Comment by alshan** [comment]
Hi @OSuwaidi, yes absolutely, we appreciate suggestions that let us better understand needs of chart creators.

 #### waterfall
The geometry itself looks like simplified [crossbar](https://lets-plot.org/pages/api/lets_plot.geom_crossbar.html). Would it have any default statistical transformation?

#### parallel coordinates
An interesting challenge. We can have it as a moonshot goal for now)

#### radar plots
Look cool! Could you create an issue?

**Reply by OSuwaidi** [reply]
That's awesome news!

Regarding the **waterfall** plot, it doesn't need a statistical "transformation" per se, rather it takes the difference operator on the y values while preserving the initial and final values as such:
`
x = [2020, 2021, 2022, 2023, 2024]
`
`
y = [100, 105, 102, 100, 105] -> y = [100, 5, -3, -2, 5, 105]
`
The key here is to highlight the **changes** in the y values as (colored) signed differences, **relative** (aligned) to the previous y value (or previous bar), and at the end, displaying the net value of those differences.

And I'll create an issue for the radar plot now.

Much appreciated!

**Comment by OSuwaidi** [comment]
Another plot idea gaining traction nowadays to consider supporting is what's called a [sankey diagram](https://sankeymatic.com/).

**Reply by alshan** [reply]
Thanx, added a feature request: #984

BTW, are you aware of this new book [Coding for Economists](https://aeturrell.github.io/coding-for-economists/intro.html)?
It features lets-plot in many places and even contains an entire chapter about lets-plot: [Easy Data Visualisation for Tidy Data with Lets-Plot](https://aeturrell.github.io/coding-for-economists/vis-letsplot.html#easy-data-visualisation-for-tidy-data-with-lets-plot).

**Comment by OSuwaidi** [comment]
Hello, another chart suggestion that is visually appealing, intuitive, practical and good for presentations is the [waffle chart](https://python-graph-gallery.com/web-waffle-chart-with-groups-evolution/).

It's roughly a stacked bar chart but with each stacked bar being quantized into individual units/grids representing its quantity (amount) relative to the total (whole).

There's a [Python package](https://github.com/gyli/PyWaffle) that implements this for reference.

**Reply by alshan** [reply]
Waffle chart is interesting, thank you, I've created a feature request: #1336
